File: Leetcode - C++/26_array_removeDuplicates.cpp
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/
// Time : O(n), SPace : O(1)

#include <iostream>
#include <vector>
#include <unordered_set>

using namespace std; 

int removeDuplicates(vector<int>& nums) {
	int i = 0;
	for(int j = 1; j < nums.size(); j++){
		if(nums[j] != nums[i]){
			i++;
			nums[i] = nums[j];
		}
	}
	return i;
}

int main() {
	vector<int> nums = {1,1,2};
	cout << "Length of array : " << removeDuplicates(nums) << endl;
}

File: Leetcode - Python/26_array_removeDuplicates.py
# https://leetcode.com/problems/remove-duplicates-from-sorted-array/

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        size = len(nums)
        if size == 0:
            return 0
        i = 0

        for j in range(1, size):
            if nums[i] != nums[j]:
                i += 1
                nums[i] = nums[j]

        return i + 1
